### PR TITLE
[SPARK-49083][CONNECT] Allow from_xml and from_json to natively work with json schemas

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?
We allow the `JsonToStructs` and `XmlToStructs` expressions to use a json schema.

### Why are the changes needed?
A couple of reasons:
- We want to use a reference to the `from_json` and `from_xml` methods in the Column API in order to make unification of the Classic and Connect Scala clients possible.
- Reduce the amount of duplication between the Function API and the SparkConnectPlanner.
- Make DataFrame and SQL API behave the same.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by hvanhovell** [comment]
@HyukjinKwon should this go in the migration guide?

**Review by HyukjinKwon — APPROVED** [review]
Seems fine cc @zhengruifeng

**Review by zhengruifeng — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
Could you resolve the conflicts and make CIs pass?

**Comment by hvanhovell** [comment]
@dongjoon-hyun yeah, working on it.

File: sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala [hunk 1]
```diff
@@ -1204,42 +1204,32 @@ class JsonFunctionsSuite extends QueryTest with SharedSparkSession {
     val df = Seq("""{"a":1}""").toDF("json")
     val invalidJsonSchema = """{"fields": [{"a":123}], "type": "struct"}"""
     checkError(
-      exception = intercept[SparkIllegalArgumentException] {
+      exception = intercept[AnalysisException] {
```
Comment (hvanhovell) [inline sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala:1207]: These changes are a result of the parsing order. The only thing we can do to make this better is to homogenise schema parsing errors.

Comment (hvanhovell) [reply sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala:1207]: Another option would be that we pass a schema format using options.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**